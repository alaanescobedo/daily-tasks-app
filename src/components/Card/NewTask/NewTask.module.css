.super_container {
    position: relative;
    /* overflow: hidden; */
    height: 70vh;
}

.container {
    height: 55px;
    background-color: rgb(var(--secondary), var(--opacity-1));
    border: 2px solid rgb(var(--secondary));
    border-radius: 10px;
    padding: 20px;
    grid-template-columns: 1fr 1fr;
    justify-items: center;
    align-content: center;
    display: flex;
    flex-direction: column;
    height: inherit;
}

.heading {
    color: rgb(var(--highlight));
    font-size: 1.6rem;
    margin-bottom: .6rem;
}

label {
    color: rgb(var(--text));
}

option {
    direction: rtl;
}

.form {
    display: grid;
    gap: 1.2rem .6rem;
    grid-template-columns: repeat(16, 1fr);
    margin-bottom: auto;
}

.input_task {
    grid-column: 1 / -1;
    border: 2px solid rgb(var(--primary));
    background-color: rgb(var(--secondary), var(--opacity-1));
    border-radius: 10px;
    height: 2.6rem;
    max-height: calc(2.6rem * 2);
    width: 100%;
    padding: 0.6rem 1rem;
    box-sizing: border-box;
    color: rgb(var(--text));
    font-family: inherit;
    font-size: inherit;
    overflow-y: hidden;
    margin: 0;
}

.input_task:focus {
    outline: none;
    border: 2px solid rgb(var(--highlight));
}

.input_date_day,
.input_date_hour,
.input_priority,
.input_recurrent {
    border: 2px solid rgb(var(--tertiary));
    background-color: rgba(var(--background), var(--opacity-7));
    border-radius: 10px;
    height: 2rem;
    color: rgba(var(--tertiary), var(--opacity-7));
    font-weight: 600;
}

.group_date_day {
    grid-column: 6 / 12;
    text-align: right;
}

.input_date_day {
    text-align: center;
}

.group_date_hour {
    grid-column: 12 / -1;
    text-align: right;
}

.input_date_hour {
    text-align: center;
    box-sizing: border-box;
    padding: 0 .4rem;
    position: relative;
}

.input_date_hour::-webkit-calendar-picker-indicator {
    position: absolute;
    width: 100%;
    height: 100%;
    transform: translateX(-10px);
    opacity: 0;
}

.group_recurrent {
    grid-column: 8 / 11;
    display: flex;
    flex-direction: column;
    align-items: flex-end
}

.input_recurrent {
    width: 2rem;
    box-sizing: border-box;
}

.input_recurrent svg {
    fill: rgba(var(--tertiary), var(--opacity-7));
    transform: scale(0.8);
}

.input_recurrent .checkbox {
    display: none;
}

.group_priority {
    grid-column: 11 / -1;
    display: flex;
    flex-direction: column;
    text-align: right;
}

.input_priority {
    text-align: center;
    box-sizing: border-box;
}

.input_submit {
    grid-column: 4 / 14;
    height: 2.6rem;
    border-radius: 10px;
    border: 2px solid rgb(var(--background));
    background-color: rgb(var(--primary));
    font-weight: 600;
    color: rgba(var(--background), var(--opacity-7));
}

.return_container {
    display: inherit;
    position: absolute;
}

.return {
    position: absolute;
    bottom: 40px;
    background-color: rgb(var(--background));
    color: rgb(var(--tertiary));
    font-size: 1.6rem;
    padding: .6rem 1.6rem;
    border-radius: 0 10px 10px 0;
    border: 2px solid rgb(var(--primary));
    border-left: none;
    transform: translateX(-.3px);
}