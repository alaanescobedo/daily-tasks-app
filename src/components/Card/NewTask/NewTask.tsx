import { ChangeEvent, FormEvent, useEffect, useState } from 'react'
import { AppContainerLayout } from '../../../layouts/AppContainerLayout'
import { getCurrentDate } from '../../../utils/getCurrentDate'
import styles from './NewTask.module.css'

const CheckIcon = (): JSX.Element => (
  <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 512 512'>
    <path d='M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z' />
  </svg>
)

const weekDaysArr = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']

export const NewTask = (): JSX.Element => {
  const [textValue, setTextValue] = useState('')
  const [checked, setChecked] = useState(false)
  const [weekDays, setWeekDays] = useState([] as string[])

  const handleSubmit = (e: FormEvent<HTMLFormElement>): void => {
    e.preventDefault()
  }

  useEffect(() => {
    const currentDay = getCurrentDate().split(',')[0]
    const indexOfDay = weekDaysArr.indexOf(currentDay)
    const sortedWeekDays = ['Today', 'Tomorrow'].concat(weekDaysArr.slice(indexOfDay + 2)).concat(weekDaysArr.slice(0, indexOfDay))

    setWeekDays(sortedWeekDays)
  }, [])

  const handleTextArea = (e: ChangeEvent<HTMLTextAreaElement>): void => {
    const textarea = e.currentTarget

    if (textarea.value.includes('\n')) return

    textarea.style.height = '2.6rem'
    setTextValue(textarea.value)

    if (textarea.scrollHeight < textarea.offsetHeight) return

    const scrollHeight = e.currentTarget.scrollHeight
    e.currentTarget.style.height = `${scrollHeight}px`
  }

  const handleCheckbox = (e: any): void => {
    setChecked(!checked)
  }

  return (
    <AppContainerLayout>
      <h1 className={styles.heading}>New task</h1>
      <form className={styles.form} onSubmit={handleSubmit}>

        <textarea
          onChange={handleTextArea}
          className={styles.input_task}
          placeholder='Task'
          maxLength={70}
          value={textValue}
        />

        <div className={styles.group_date_day}>
          <select name='date_day' id='date_day' className={styles.input_date_day} onChange={handleSelectChange}>
            {weekDays.map((day, index) => (
              <option key={index} value={index + 1}>{day}</option>
            ))}
          </select>
          <label htmlFor='date_day'>Day</label>
        </div>

        <div className={styles.group_date_hour}>
          <input name='date_hour' id='date_hour' type='time' className={styles.input_date_hour} />
          <label htmlFor='date_hour'>Hour</label>
        </div>

        <div className={styles.group_recurrent}>
          <span className={styles.input_recurrent} onClick={handleCheckbox}>
            {checked ? <CheckIcon /> : ''}
            <input name='recurrent' id='recurrent' type='checkbox' className={styles.checkbox} checked={checked} readOnly />
          </span>
          <label htmlFor='recurrent'>Recurrent</label>
        </div>

        <div className={styles.group_priority}>
          <select name='priority' id='priority' className={styles.input_priority} defaultValue='high'>
            <option value='low'>Low</option>
            <option value='medium'>Medium</option>
            <option value='high'>High</option>
          </select>
          <label htmlFor='priority'>Priority</label>
        </div>

        <button className={styles.input_submit}>Create</button>
      </form>
    </AppContainerLayout>
  )
}
