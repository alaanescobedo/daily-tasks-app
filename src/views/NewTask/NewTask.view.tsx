import { NEW_TASK_INPUT_CONFIG } from '@config/new-task.config'
import { useForm } from '@hooks'
import { NewTaskFormLayout } from '@layouts'
import { Checkbox, Time, InputGroup, Select, TextArea, WeekdaysId, WeekdaysLabel } from '@components/Form'

import styles from './NewTask.module.css'
import { getSevenDays } from '@utils/getSevenDays'
import { getCurrentTimePlus5minutes } from '@utils/getCurrentTimePlus5minutes'
import { getCurrentDate } from '@utils/getCurrentDate'

const weekdays = getSevenDays() as Array<{ id: WeekdaysId, label: WeekdaysLabel }>

export const NewTaskView = (): JSX.Element => {
  const { fields, handleFieldsChange, handleSubmit, errors } = useForm(NEW_TASK_INPUT_CONFIG)
  const { title, day, hour, recurrent, priority } = fields

  const minTime = getCurrentDate().split(',')[2].split(':').slice(0, 2).join(':').trim()

  return (
    <NewTaskFormLayout sendLabel='Create Task' handleSubmit={handleSubmit} id='newTask'>
      <div style={{ gridColumn: '1 / -1', color: 'tomato', backgroundColor: 'rebeccapurple', padding: '0 1rem' }}>
        <p>{errors.title.errorMessage}</p>
        <p>{errors.day.errorMessage}</p>
        <p>{errors.hour.errorMessage}</p>
        <p>{errors.recurrent.errorMessage}</p>
        <p>{errors.priority.errorMessage}</p>
      </div>
      <TextArea value={title.value} handleChange={handleFieldsChange} placeholder={title.placeholder} id={title.id} />

      <InputGroup label={day.label} htmlFor={day.id} className={styles.group_date_day}>
        <Select options={weekdays} id={day.id} handleChange={handleFieldsChange} defaultValue={day.defaultValue} />
      </InputGroup>

      <InputGroup label='Hour' htmlFor={hour.id} className={styles.group_date_hour}>
        <Time id={hour.id} handleChange={handleFieldsChange} defaultValue={getCurrentTimePlus5minutes()} minTime={minTime} />
      </InputGroup>

      <InputGroup label='Recurrent' htmlFor={recurrent.id} className={styles.group_recurrent}>
        <Checkbox id={recurrent.id} checked />
      </InputGroup>

      <InputGroup label='Priority' htmlFor={priority.id} className={styles.group_priority}>
        <Select options={priority.options} id={priority.id} handleChange={handleFieldsChange} defaultValue={priority.defaultValue} />
      </InputGroup>
    </NewTaskFormLayout>
  )
}
