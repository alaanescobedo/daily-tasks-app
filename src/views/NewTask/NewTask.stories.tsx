import { ComponentStory, ComponentMeta } from '@storybook/react'
import { NewTaskView as NewTaskViewComponent } from './NewTask.view'

const NewTaskViewStory: ComponentMeta<typeof NewTaskViewComponent> = {
  title: 'Views/New Task',
  component: NewTaskViewComponent
}
export default NewTaskViewStory

const Template: ComponentStory<typeof NewTaskViewComponent> = () => <NewTaskViewComponent />

export const NewTask = Template.bind({})
