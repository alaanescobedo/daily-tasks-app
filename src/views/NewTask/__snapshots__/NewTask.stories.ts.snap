// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Views/New Task New Task 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <div
      className="container"
    >
      <h4
        className="username"
      >
        Guest
      </h4>
      <h4
        className="date"
      >
        Tuesday,  4/12/2022
        <div
          style={
            Object {
              "display": "none",
            }
          }
        >
          Clock
        </div>
      </h4>
    </div>
    <div
      className="super_container"
    >
      <div
        className="container"
      >
        <h1
          className="heading"
        >
          New Task
        </h1>
        <form
          className="form"
          onSubmit={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": "rebeccapurple",
                "color": "tomato",
                "gridColumn": "1 / -1",
                "padding": "0 1rem",
              }
            }
          >
            <p>
              
            </p>
            <p>
              
            </p>
            <p>
              
            </p>
            <p>
              
            </p>
            <p>
              
            </p>
          </div>
          <textarea
            className="input_task"
            id="title"
            maxLength={70}
            name="title"
            onChange={[Function]}
            placeholder="Title"
            value=""
          />
          <div
            className="group_date_day"
          >
            <select
              className="input_select"
              defaultValue="2022-04-13"
              id="day"
              name="day"
              onChange={[Function]}
            >
              <option
                value="4/12/2022"
              >
                Today
              </option>
              <option
                value="4/13/2022"
              >
                Tomorrow
              </option>
              <option
                value="4/14/2022"
              >
                Thursday
              </option>
              <option
                value="4/15/2022"
              >
                Friday
              </option>
              <option
                value="4/16/2022"
              >
                Saturday
              </option>
              <option
                value="4/17/2022"
              >
                Sunday
              </option>
              <option
                value="4/18/2022"
              >
                Monday
              </option>
            </select>
            <label
              htmlFor="day"
            >
              Day
            </label>
          </div>
          <div
            className="group_date_hour"
          >
            <input
              className="input_date_hour"
              defaultValue="23:56"
              id="hour"
              min="23:51"
              name="hour"
              onChange={[Function]}
              type="time"
            />
            <label
              htmlFor="hour"
            >
              Hour
            </label>
          </div>
          <div
            className="group_recurrent"
          >
            <span
              className="input_recurrent"
              onClick={[Function]}
            >
              <svg
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                />
              </svg>
              <input
                checked={true}
                className="checkbox"
                id="recurrent"
                name="recurrent"
                readOnly={true}
                type="checkbox"
              />
            </span>
            <label
              htmlFor="recurrent"
            >
              Recurrent
            </label>
          </div>
          <div
            className="group_priority"
          >
            <select
              className="input_select"
              defaultValue="medium"
              id="priority"
              name="priority"
              onChange={[Function]}
            >
              <option
                value="low"
              >
                Low
              </option>
              <option
                value="medium"
              >
                Medium
              </option>
              <option
                value="high"
              >
                High
              </option>
            </select>
            <label
              htmlFor="priority"
            >
              Priority
            </label>
          </div>
          <button
            className="input_submit"
            type="submit"
          >
            Create Task
          </button>
        </form>
      </div>
      <div
        className="return_container"
      >
        <a
          aria-current="page"
          className="return active"
          href="/"
          onClick={[Function]}
        >
          Return
        </a>
      </div>
    </div>
     
  </div>,
  " ",
]
`;
